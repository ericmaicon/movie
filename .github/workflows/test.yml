name: Lint & Test

on: push

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v2
        with:
          node-version: '14'
      - name: Cache dependencies
        uses: actions/cache@v2.1.4
        id: cache-dependencies
        with:
          path: |
            node_modules
            */*/node_modules
          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
      - name: Install Dependencies
        if: steps.cache-dependencies.outputs.cache-hit != 'true'
        run: yarn install && yarn bootstrap
      - name: Link
        run: yarn lerna link
      - name: Build deps
        run: yarn build
      - name: Lint
        run: yarn lint
      - name: Run Tests
        run: yarn test
      - uses: codecov/codecov-action@v1
        with:
          token: ${{ secrets.CODECOV_TOKEN }}
          directory: packages/**/coverage
          fail_ci_if_error: true
