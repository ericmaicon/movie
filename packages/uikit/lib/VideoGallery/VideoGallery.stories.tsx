import React from 'react';

import { Story, Meta } from '@storybook/react';

import VideoGallery, { VideoGalleryProps } from './VideoGallery';

export default {
  title: 'Layout/VideoGallery',
  component: VideoGallery,
} as Meta;

const Template: Story<VideoGalleryProps> = (args) => <VideoGallery {...args} />;

export const Default = Template.bind({});
Default.args = {
  items: [
    {
      id: 1,
      image:
        'https://www.themoviedb.org/t/p/w220_and_h330_face/drulhSX7P5TQlEMQZ3JoXKSDEfz.jpg',
      title:
        'When the crew of a space junk collector ship called The Victory discovers a humanoid..',
      link: '',
    },
    {
      id: 2,
      image:
        'https://www.themoviedb.org/t/p/w220_and_h330_face/srYya1ZlI97Au4jUYAktDe3avyA.jpg',
      title:
        'When the crew of a space junk collector ship called The Victory discovers a humanoid..',
      link: '',
    },
    {
      id: 3,
      image:
        'https://www.themoviedb.org/t/p/w220_and_h330_face/6TPZSJ06OEXeelx1U1VIAt0j9Ry.jpg',
      title:
        'When the crew of a space junk collector ship called The Victory discovers a humanoid..',
      link: '',
    },
    {
      id: 4,
      image:
        'https://www.themoviedb.org/t/p/w220_and_h330_face/dWSnsAGTfc8U27bWsy2RfwZs0Bs.jpg',
      title:
        'When the crew of a space junk collector ship called The Victory discovers a humanoid..',
      link: '',
    },
    {
      id: 5,
      image:
        'https://www.themoviedb.org/t/p/w220_and_h330_face/vfuzELmhBjBTswXj2Vqxnu5ge4g.jpg',
      title:
        'When the crew of a space junk collector ship called The Victory discovers a humanoid..',
      link: '',
    },
    {
      id: 6,
      image:
        'https://www.themoviedb.org/t/p/w220_and_h330_face/c7VlGCCgM9GZivKSzBgzuOVxQn7.jpg',
      title:
        'When the crew of a space junk collector ship called The Victory discovers a humanoid..',
      link: '',
    },
    {
      id: 7,
      image:
        'https://www.themoviedb.org/t/p/w220_and_h330_face/lOSdUkGQmbAl5JQ3QoHqBZUbZhC.jpg',
      title:
        'When the crew of a space junk collector ship called The Victory discovers a humanoid..',
      link: '',
    },
    {
      id: 8,
      image:
        'https://www.themoviedb.org/t/p/w220_and_h330_face/6XYLiMxHAaCsoyrVo38LBWMw2p8.jpg',
      title:
        'When the crew of a space junk collector ship called The Victory discovers a humanoid..',
      link: '',
    },
    {
      id: 9,
      image:
        'https://www.themoviedb.org/t/p/w220_and_h330_face/h9DIlghaiTxbQbt1FIwKNbQvEL.jpg',
      title:
        'When the crew of a space junk collector ship called The Victory discovers a humanoid..',
      link: '',
    },
  ],
};
