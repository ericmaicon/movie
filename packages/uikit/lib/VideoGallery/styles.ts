import styled from 'styled-components';

/**
 * HL: Highlight image
 * RE: Regular image
 */
export const VideoGallery = styled.div`
  display: grid;
  grid-template-columns: repeat(6, 1fr);
  grid-gap: 15px;

  img:first-child {
    grid-column-start: 1;
    grid-column-end: 3;
    grid-row-start: 1;
    grid-row-end: 3;
  }
`;
