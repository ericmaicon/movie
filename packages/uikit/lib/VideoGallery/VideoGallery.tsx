import React from 'react';

import { VideoCard } from '~/VideoCard/styles';

import * as Styled from './styles';

export interface Item {
  id: string | number;
  image: string;
  title: string;
}

export interface VideoGalleryProps {
  items: Item[];
  onSelectItem: (id: string | number) => void;
}

const VideoGallery = ({ items, onSelectItem, ...rest }: VideoGalleryProps) => {
  return (
    <Styled.VideoGallery {...rest}>
      {items.map((item: Item) => (
        <VideoCard
          data-testid="video-card"
          key={item.id}
          src={item.image}
          alt={item.title}
          onClick={() => onSelectItem(item.id)}
        />
      ))}
    </Styled.VideoGallery>
  );
};

export default VideoGallery;
