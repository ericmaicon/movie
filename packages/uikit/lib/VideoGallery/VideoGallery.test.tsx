import React from 'react';

import { render } from '~/test/index';

import VideoGallery from './VideoGallery';

const items = [
  {
    id: 1,
    image:
      'https://www.themoviedb.org/t/p/w220_and_h330_face/drulhSX7P5TQlEMQZ3JoXKSDEfz.jpg',
    title:
      'When the crew of a space junk collector ship called The Victory discovers a humanoid..',
    link: '',
  },
  {
    id: 2,
    image:
      'https://www.themoviedb.org/t/p/w220_and_h330_face/srYya1ZlI97Au4jUYAktDe3avyA.jpg',
    title:
      'When the crew of a space junk collector ship called The Victory discovers a humanoid..',
    link: '',
  },
  {
    id: 3,
    image:
      'https://www.themoviedb.org/t/p/w220_and_h330_face/6TPZSJ06OEXeelx1U1VIAt0j9Ry.jpg',
    title:
      'When the crew of a space junk collector ship called The Victory discovers a humanoid..',
    link: '',
  },
  {
    id: 4,
    image:
      'https://www.themoviedb.org/t/p/w220_and_h330_face/dWSnsAGTfc8U27bWsy2RfwZs0Bs.jpg',
    title:
      'When the crew of a space junk collector ship called The Victory discovers a humanoid..',
    link: '',
  },
  {
    id: 5,
    image:
      'https://www.themoviedb.org/t/p/w220_and_h330_face/vfuzELmhBjBTswXj2Vqxnu5ge4g.jpg',
    title:
      'When the crew of a space junk collector ship called The Victory discovers a humanoid..',
    link: '',
  },
  {
    id: 6,
    image:
      'https://www.themoviedb.org/t/p/w220_and_h330_face/c7VlGCCgM9GZivKSzBgzuOVxQn7.jpg',
    title:
      'When the crew of a space junk collector ship called The Victory discovers a humanoid..',
    link: '',
  },
  {
    id: 7,
    image:
      'https://www.themoviedb.org/t/p/w220_and_h330_face/lOSdUkGQmbAl5JQ3QoHqBZUbZhC.jpg',
    title:
      'When the crew of a space junk collector ship called The Victory discovers a humanoid..',
    link: '',
  },
  {
    id: 8,
    image:
      'https://www.themoviedb.org/t/p/w220_and_h330_face/6XYLiMxHAaCsoyrVo38LBWMw2p8.jpg',
    title:
      'When the crew of a space junk collector ship called The Victory discovers a humanoid..',
    link: '',
  },
  {
    id: 9,
    image:
      'https://www.themoviedb.org/t/p/w220_and_h330_face/h9DIlghaiTxbQbt1FIwKNbQvEL.jpg',
    title:
      'When the crew of a space junk collector ship called The Victory discovers a humanoid..',
    link: '',
  },
];

describe('VideoGallery', () => {
  it('should render correctly', () => {
    const { getByTestId } = render(
      <VideoGallery data-testid="video-galery" items={items} />,
    );

    const videoGalleryElement = getByTestId('video-galery');

    expect(videoGalleryElement).toBeInTheDocument();
    expect(videoGalleryElement.childElementCount).toBe(9);
    expect(videoGalleryElement).toMatchSnapshot();
  });
});
