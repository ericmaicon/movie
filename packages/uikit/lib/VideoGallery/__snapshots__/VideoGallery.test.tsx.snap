// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoGallery should render correctly 1`] = `
.c1 {
  cursor: pointer;
  width: 100%;
}

.c1:hover {
  box-shadow: 0px 0px 0px 2px rgba(0,0,0,0.2);
}

.c0 {
  display: grid;
  grid-template-columns: repeat(6,1fr);
  grid-gap: 15px;
}

.c0 img:first-child {
  grid-column-start: 1;
  grid-column-end: 3;
  grid-row-start: 1;
  grid-row-end: 3;
}

<div
  class="c0"
  data-testid="video-galery"
>
  <img
    alt="When the crew of a space junk collector ship called The Victory discovers a humanoid.."
    class="c1"
    data-testid="video-card"
    src="https://www.themoviedb.org/t/p/w220_and_h330_face/drulhSX7P5TQlEMQZ3JoXKSDEfz.jpg"
  />
  <img
    alt="When the crew of a space junk collector ship called The Victory discovers a humanoid.."
    class="c1"
    data-testid="video-card"
    src="https://www.themoviedb.org/t/p/w220_and_h330_face/srYya1ZlI97Au4jUYAktDe3avyA.jpg"
  />
  <img
    alt="When the crew of a space junk collector ship called The Victory discovers a humanoid.."
    class="c1"
    data-testid="video-card"
    src="https://www.themoviedb.org/t/p/w220_and_h330_face/6TPZSJ06OEXeelx1U1VIAt0j9Ry.jpg"
  />
  <img
    alt="When the crew of a space junk collector ship called The Victory discovers a humanoid.."
    class="c1"
    data-testid="video-card"
    src="https://www.themoviedb.org/t/p/w220_and_h330_face/dWSnsAGTfc8U27bWsy2RfwZs0Bs.jpg"
  />
  <img
    alt="When the crew of a space junk collector ship called The Victory discovers a humanoid.."
    class="c1"
    data-testid="video-card"
    src="https://www.themoviedb.org/t/p/w220_and_h330_face/vfuzELmhBjBTswXj2Vqxnu5ge4g.jpg"
  />
  <img
    alt="When the crew of a space junk collector ship called The Victory discovers a humanoid.."
    class="c1"
    data-testid="video-card"
    src="https://www.themoviedb.org/t/p/w220_and_h330_face/c7VlGCCgM9GZivKSzBgzuOVxQn7.jpg"
  />
  <img
    alt="When the crew of a space junk collector ship called The Victory discovers a humanoid.."
    class="c1"
    data-testid="video-card"
    src="https://www.themoviedb.org/t/p/w220_and_h330_face/lOSdUkGQmbAl5JQ3QoHqBZUbZhC.jpg"
  />
  <img
    alt="When the crew of a space junk collector ship called The Victory discovers a humanoid.."
    class="c1"
    data-testid="video-card"
    src="https://www.themoviedb.org/t/p/w220_and_h330_face/6XYLiMxHAaCsoyrVo38LBWMw2p8.jpg"
  />
  <img
    alt="When the crew of a space junk collector ship called The Victory discovers a humanoid.."
    class="c1"
    data-testid="video-card"
    src="https://www.themoviedb.org/t/p/w220_and_h330_face/h9DIlghaiTxbQbt1FIwKNbQvEL.jpg"
  />
</div>
`;
